### PR TITLE
add scheduled pipelines with custom cluster for CNI benchmarks

### PR DESCRIPTION
**Pull request by jshr-w**
This PR adds 2 new pipelines, to add alternate versions of the existing Service Churn and Cluster Churn pipelines for CNI benchmarking -- these pipelines are intended only for use with Cilium, and skip any steps related to cluster management.

- Add module to control initial and final number of nodes per user nodepool (To scale up before test and scale down after test, where necessary)
- Add test_type field to result
- Skip RG create/delete when SKIP_RESOURCE_MANAGEMENT is true

### CODE REVIEW

File: jobs/competitive-test.yml [hunk 1]
```diff
@@ -48,6 +48,9 @@ parameters:
 - name: ssh_key_enabled
   type: boolean
   default: true
+- name: initial_nodes_per_nodepool
```
Comment (alyssa1303) [inline jobs/competitive-test.yml:51]: Can we put this as part of matrix since this variable is not shared between pipelines but only specific to these 2 pipelines and remove it from here?

File: jobs/competitive-test.yml [hunk 2]
```diff
@@ -104,3 +108,4 @@ jobs:
       terraform_arguments: ${{ parameters.terraform_arguments }}
       retry_attempt_count: ${{ parameters.retry_attempt_count }}
       credential_type: ${{ parameters.credential_type }}
+      topology: ${{ parameters.topology }}
```
Comment (alyssa1303) [inline jobs/competitive-test.yml:111]: Remove

File: jobs/competitive-test.yml [hunk 3]
```diff
@@ -82,6 +85,7 @@ jobs:
       regions: ${{ parameters.regions }}
       topology: ${{ parameters.topology }}
       engine: ${{ parameters.engine }}
+      nodes_per_nodepool: ${{ parameters.initial_nodes_per_nodepool }}
```
Comment (alyssa1303) [inline jobs/competitive-test.yml:88]: Remove

File: steps/cleanup-resources.yml [hunk 4]
```diff
@@ -13,6 +13,9 @@ parameters:
 - name: credential_type
   type: string
   default: managed_identity
+- name: topology
```
Comment (alyssa1303) [inline steps/cleanup-resources.yml:16]: Remove here. We should do this as part of collect step under topology folder since this is only specific to these 2 pipelines

File: steps/cleanup-resources.yml [hunk 5]
```diff
@@ -30,4 +33,9 @@ steps:
     echo "Delete resource group $RUN_ID"
     az group delete --name $RUN_ID --yes
   displayName: "Destroy Resource Group"
-  condition: ${{ eq(parameters.cloud, 'azure') }}
+  condition: and(${{ eq(parameters.cloud, 'azure') }}, ne(variables['SKIP_RESOURCE_MANAGEMENT'], 'true'))
+
+- ${{ if eq(parameters.topology, 'cilium-usercluster') }}:
+  - template: /steps/topology/${{ parameters.topology }}/cleanup-resources.yml@self
+    parameters:
+      regions: ${{ parameters.regions }}
```
Comment (alyssa1303) [inline steps/cleanup-resources.yml:41]: move this to `collect-clusterloader2.yaml` under `topology` folder

File: steps/topology/cilium-usercluster-autoscale/cleanup-resources.yml [hunk 6]
```diff
@@ -0,0 +1,11 @@
+parameters:
```
Comment (alyssa1303) [inline steps/topology/cilium-usercluster-autoscale/cleanup-resources.yml:1]: Just move this to collect-clusterloader2.yaml. No need for a new file

File: steps/topology/cilium-usercluster/validate-resources.yml [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+steps:
+  - template: /steps/cloud/${{ parameters.cloud }}/update-kubeconfig.yml
+    parameters:
+      role: ces
+      region: ${{ parameters.regions[0] }}
+  - template: /steps/engine/clusterloader2/cilium/scale-cluster.yml
+    parameters:
+      role: ces
+      region: ${{ parameters.regions[0] }}
+      nodes_per_nodepool: ${{ parameters.nodes_per_nodepool }}
```
Comment (alyssa1303) [inline steps/topology/cilium-usercluster/validate-resources.yml:22]: Can we just say `50` here directly? Each topology already have its own validate-resources.yml so there's isn't really a need to use parameter

File: pipelines/perf-eval/slo-servicediscovery-ces.yml [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+              cilium_enabled: True
+              network_policy: cilium
+              network_dataplane: cilium
+              service_test: True
+              cl2_config_file: load-config.yaml
+          max_parallel: 2
+          timeout_in_minutes: 360
+          credential_type: service_connection
+          ssh_key_enabled: false
+          initial_nodes_per_nodepool: 50
```
Comment (alyssa1303) [inline pipelines/perf-eval/slo-servicediscovery-ces.yml:46]: remove here since we can just use 50 directly in `validate-resources.yml`

File: steps/topology/cilium-usercluster-autoscale/validate-resources.yml [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+steps:
+  - template: /steps/cloud/${{ parameters.cloud }}/update-kubeconfig.yml
+    parameters:
+      role: ces
+      region: ${{ parameters.regions[0] }}
+  - template: /steps/engine/clusterloader2/cilium/scale-cluster.yml
+    parameters:
+      role: ces
+      region: ${{ parameters.regions[0] }}
+      nodes_per_nodepool: ${{ parameters.nodes_per_nodepool }}
```
Comment (alyssa1303) [inline steps/topology/cilium-usercluster-autoscale/validate-resources.yml:20]: same here we can just use number

File: steps/topology/cilium-usercluster/cleanup-resources.yml [hunk 10]
```diff
@@ -0,0 +1,11 @@
+parameters:
```
Comment (alyssa1303) [inline steps/topology/cilium-usercluster/cleanup-resources.yml:1]: same comment. just move to collect-clusterloader2.yaml so we don't need to touch `competitive-test.yml`

File: steps/validate-resources.yml [hunk 11]
```diff
@@ -7,10 +7,14 @@ parameters:
   type: string
 - name: engine
   type: string
+- name: nodes_per_nodepool
```
Comment (alyssa1303) [inline steps/validate-resources.yml:10]: undo this change

File: modules/python/clusterloader2/slo/config/cluster-scale-config.yaml [hunk 12]
```diff
@@ -95,13 +90,6 @@ steps:
           labelSelector: slo = true
           timeout: {{$operationTimeout}}
           refreshInterval: 1m
-  - name: Waiting for latency pods to be running
```
Comment (alyssa1303) [inline modules/python/clusterloader2/slo/config/cluster-scale-config.yaml:98]: Why is it removed? We need this to wait for pod ready before adding the next scale step so we can scale gradually

File: modules/python/clusterloader2/slo/config/cluster-scale-config.yaml [hunk 13]
```diff
@@ -115,26 +103,18 @@ steps:
         objectBundle:
           - basename: deployment-churn-{{$j}}
             objectTemplatePath: deployment_template.yaml
-  - name: Waiting for latency pods to be deleted
```
Comment (alyssa1303) [inline modules/python/clusterloader2/slo/config/cluster-scale-config.yaml:118]: Same here, why is it removed? We'd like to delete gradually to not overload the cluster

File: pipelines/perf-eval/cilium-cluster-churn-nodes-ces.yml
Comment (alyssa1303) [inline pipelines/perf-eval/cilium-cluster-churn-nodes-ces.yml:1]: I'm a bit confused but why do we need 2 files for cluster-churn and 2 files for slo-servicediscovery? We only need 1 file for each scenario for the internal testing to see if there's any improvement right? The official feature should already be cover by 2 existing pipelines.

File: steps/validate-resources.yml [hunk 14]
```diff
@@ -14,3 +14,4 @@ steps:
     cloud: ${{ parameters.cloud }}
     regions: ${{ parameters.regions }}
     engine: ${{ parameters.engine }}
+    topology: ${{ parameters.topology }}
```
Comment (alyssa1303) [inline steps/validate-resources.yml:17]: remove since this is not needed anymore

File: modules/python/clusterloader2/slo/config/cluster-scale-config.yaml (hunk 12 above)
Comment (jshr-w) [reply modules/python/clusterloader2/slo/config/cluster-scale-config.yaml:98]: Added back, had made a temporary change to try refactoring but I think I'll punt that to a future PR

File: pipelines/perf-eval/cilium-cluster-churn-nodes-ces.yml
Comment (jshr-w) [reply pipelines/perf-eval/cilium-cluster-churn-nodes-ces.yml:1]: Discussed offline. TLDR for future readers; -ces pipelines are temporary for comparison with long-running pipelines, -feature pipeline will be manually run for adding features

**Review by alyssa1303 — APPROVED** [review]
